handlers/security.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'httparty'
+
+module Lita
+  module Handlers
+    # some comment
+    class Security < Handler
+      config :github, default: Zooniverse::Github.new
+
+      route(/^(dependabot)\s*(.*)/, :get_dependabot_issues, command: true,
+                                                            help: { 'status dependabot' => 'displays dependabot security alerts' })
+
+      def get_dependabot_issues(response)
+        get_issues = true
+        last_repo_listed = nil
+        alerts = []
+        repo_to_alert_count = {}
+
+        while get_issues == true
+          res = config.github.get_dependabot_issues(last_repo_listed)
+          if res.nil?
+            get_issues = false
+            break
+          end
+          edges = res['data']['organization']['repositories']['edges']
+          nodes = res['data']['organization']['repositories']['nodes']
+          nodes.each do |node|
+            node_alerts = node['vulnerabilityAlerts']['nodes']
+            repo_name = node['name']
+            next if repos_to_skip.include? repo_name
+            next if node_alerts.empty?
+
+            filter_alerts(node_alerts, alerts, repo_to_alert_count, repo_name)
+          end
+          repo_count = edges.length
+          last_repo_listed = edges[repo_count - 1]['cursor']
+          get_issues = false if repo_count < 100
+        end
+
+        response.reply("#{alerts.length} Alerts Total: \n #{format_alerts(repo_to_alert_count)}")
+      end
+
+      private
+
+      def filter_alerts(node_alerts, alerts, repo_to_alert_count, repo_name)
+        node_alerts.each do |alert|
+          next unless alert['dismissedAt'].nil?
+          next unless alert['fixedAt'].nil?
+
+          vulnerability = alert['securityVulnerability']
+          alerts << { repo_name => vulnerability }
+          add_alert_count(repo_to_alert_count, repo_name)
+        end
+      end
+
+      def add_alert_count(repo_to_alert_count, repo_name)
+        if repo_to_alert_count[repo_name].nil?
+          repo_to_alert_count[repo_name] = 1
+        else
+          repo_to_alert_count[repo_name] += 1
+        end
+      end
+
+      def repos_to_skip
+        %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client]
```

Comment:
maybe move these to a constant vs a method that creates a new array each time it's used in the looping structure. Alternatively memoize the array on the method call to avoid the extra setup costs (heap space and cpu etc)
```
# memoize the value via an instance var
@repos_to_skip ||= %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client]

```

while this example is minimal in terms of heap space costs etc these things can add up if you're loop becomes bigger than you expected.

Reply:
also i assume these values are always case sensitive as the source data comes from the repo name in github. What happens if we rename the repo to lowercase? these will break but we haven't really changed the name. it might be useful to compare these as all lowercase to avoid this

```
# setup all lowercase repo names
@repos_to_skip ||= %w[next-cookie-auth-panoptes Cellect science-gossip-data seven-ten Seven-Ten-Client].map(&:downcase)

# and then compare against the lowercase repo name from gh
repos_to_skip.include? repo_name.lowercase
```